task_netcore/task2/task2/Controllers/DataController.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using task2.Services;
+
+namespace task2.Controllers
+{
+    [Route("api/[controller]")]
+    public class DataController : Controller
+    {
+        private readonly IDataService dataService;
```

Comment:
According to some naming conventions, private fields should use underscore like `_dataService`. Then, you could assign value without `this`